experimental/framework/fluid-static/src/FluidStatic.ts

```diff
@@ -105,40 +37,67 @@ export class FluidInstance {
         this.containerService = getContainerService;
     }
 
-    public async createContainer(id: string, config: ContainerCreateConfig): Promise<FluidContainer> {
-        const registryEntries = this.getRegistryEntries(config.dataObjects);
+    public async createContainer(id: string, config: ContainerConfig): Promise<FluidContainer> {
+        const [registryEntries, sharedObjects] = this.parseDataObjectsFromSharedObjects(config);
         const container = await getContainer(
             this.containerService,
             id,
-            new DOProviderContainerRuntimeFactory(registryEntries, config.initialDataObjects),
+            new DOProviderContainerRuntimeFactory(registryEntries, sharedObjects, config.initialObjects),
```

Comment:
Make the container runtime factory take the ContainerConfig object and do the parsing in there.